Add lock and unlock to Comments

timendum opened the pull request:
Hi,
by moving the `lock` and `unlock` methods from Submission to ThingModerationMixin, they are available also to Comment objects.

I've added the test cases, but I'm not providing a cassette, you can easly generate one by running `pytest`.

The methods are tested anyway, on my test subreddit.

Fixes #1073

## Feature Summary and Justification Comment objects.

This feature provides lock and unlock methods for 

## References

* https://www.reddit.com/r/modnews/comments/brgr8i/moderators_you_may_now_lock_individual_comments/

bboe commented:
Thanks for the PR!

> I've added the test cases, but I'm not providing a cassette, you can easly generate one by running pytest.

Is there any specific reason why you aren't adding those?

timendum commented:
> > I've added the test cases, but I'm not providing a cassette, you can easly generate one by running pytest.
> 
> Is there any specific reason why you aren't adding those?
> 
Just for consistency, others test are on r/reddit_api_test but I can't reproduce a test on that subreddit.  
I thought you wanted to run the test case properly and I provided a comment id on reddit_api_test.  
Of course I can provide them, if it's easier for you.

I added myself to the authors file and squashed the commits.

bboe commented:
> Just for consistency, others test are on r/reddit_api_test but I can't reproduce a test on that subreddit.
> I thought you wanted to run the test case properly and I provided a comment id on reddit_api_test.
Of course I can provide them, if it's easier for you.

That makes sense, thanks for thinking of that. I'll generate them in that case.

jarhill0 commented:
> others test are on r/reddit_api_test

I don't think that's a requirement. I've recorded tests on my own testing subreddit and those have been merged without issue.

This doesn't really matter though, because bboe has said he will record the cassettes.

bboe commented:
Thanks! Merged into branch tmp, and adding my commit onto it before merging into master.
